### ISSUE TITLE
Update to NServiceBus v7?

### ISSUE THREAD

**Issue by andreasohlund** [sbmako/NServiceBus.MongoDB#29]
With v7 just right around the corner are there any plans to update the persister to support it? (there should not be any dramatic changes on the persistence seam)

**Comment by sbmako** [comment]
Yes.   I will be updating it.   I am also looking into supporting .net core.

**Comment by andreasohlund** [comment]
Awesome, let me know if you need any help, pr-reviews etc

**Comment by kpko** [comment]
Hi, can you already tell us what the current state of the implementation looks like? It would be awesome to be able to use MongoDB persistence in NServiceBus 7.

**Comment by sbmako** [comment]
There is a pre-release version available on nuget.org.   We have been using it for a short while now and have not experienced any issues.

Please let me know if you find anything

**Comment by willwolfram18** [comment]
@sbmako Are there any details on when this will move out of pre-release and into a production version of the package?

**Comment by sbmako** [comment]
As i mentioned, we have not had any issues yet but i am looking to get some feedback from other users.  Maybe by first of this month?

Thoughts?

**Comment by willwolfram18** [comment]
Is the plan to have the project target NET Standard 2.0 instead of .NET 4.6.1 when it's released?

**Comment by sbmako** [comment]
Correct.  There is already dotnet core branch that does exactly that.  There is also a pre-release nuget package on nuget.org.

**Comment by willwolfram18** [comment]
I do see the dotnet core branch, however it appears that `NServiceBus.MongoDB` pre-release package was restored using .NET Framework 4.6.1 which results in a warning in the build.
![image](https://user-images.githubusercontent.com/8053660/44166896-6e816800-a081-11e8-93de-4822d2a0bddb.png)
```
Package 'NServiceBus.MongoDB 7.0.2-nsb7.1' was restored using '.NETFramework,Version=v4.6.1' instead of the project target framework '.NETCoreApp,Version=v2.1'. This package may not be fully compatible with your project.
```

Pulling down the solution, it appears that the package is targeting NET Standard 2.0, and it doesn't appear any dependencies are targeting .NET Framework when the project is built. Perhaps it's an issue with how AppVeyor does the build step?

**Comment by sbmako** [comment]
Thanks, will take care of that.

**Comment by sbmako** [comment]
There is another pre-release package: https://www.nuget.org/packages/NServiceBus.MongoDB/7.0.2-dotnet-core.1

**Comment by willwolfram18** [comment]
Apparently I never noticed that one 😢  Sorry about that! Thanks for the help!
